opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java

```diff
@@ -443,12 +447,20 @@ public void displayToast(@StringRes int resourceId) {
         Toast.makeText(this, resourceId, Toast.LENGTH_SHORT).show();
     }
 
+    @SuppressLint("MissingPermission")
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == REQUEST_CODE_GET_JSON && resultCode == RESULT_OK && data.hasExtra(JSON_FORM_PARAM_JSON)) {
             String json = data.getStringExtra(JSON_FORM_PARAM_JSON);
             Log.d(TAG, json);
             listTaskPresenter.saveJsonForm(json);
+        } else if (requestCode == Constants.RequestCode.LOCATION_SETTINGS && hasRequestedLocation) {
```

Comment:
Why is `hasRequestedLocation` needed here?

Reply:
This is because if the baseMap i.e `KujakuMapView` one does not define` warm_gps=false` then on initializing `KujakuMapView` it will check if location services are on and will ask user to enable them. 
This will invoke `onActivityResult` depending on if user granted/declined to enable location services. In this case we wouldn't want to validate user location.

So this flag indicates that there was an explicit attempt to enable location services and its not initialization.

Reply:
Okay.